amethyst_core/src/thread_pool.rs

```diff
@@ -0,0 +1,53 @@
+use amethyst_error::Error;
+#[cfg(not(no_threading))]
+use std::sync::Arc;
+
+#[cfg(not(no_threading))]
+#[derive(Debug, Clone)]
+pub struct ThreadPool {
```

Comment:
There's not necessarily a need to wrap `Arc<rayon::ThreadPool>` but I found that otherwise there would be not so nice to read `#[cfg(` in quite a lot of places.